actionpack/test/dispatch/content_security_policy_test.rb

```diff
@@ -302,6 +302,66 @@ def test_raises_runtime_error_when_unexpected_source
   end
 end
 
+class ContentSecurityPolicyMiddlewareTest < ActiveSupport::TestCase
+  def setup
+    @env = Rack::MockRequest.env_for("", {})
+    @env["action_dispatch.content_security_policy"] = ActionDispatch::ContentSecurityPolicy.new do |p|
+      p.default_src -> { :self  }
+    end
+    @env["action_dispatch.content_security_policy_nonce_generator"] = proc { "iyhD0Yc0W+c=" }
+    @env["action_dispatch.content_security_policy_report_only"] = false
+    @env["action_dispatch.show_exceptions"] = :none
+
+    @default_csp = "default-src 'self' https: http:"
+  end
+
+  def test_rack_lint
+    app = proc { [200, {}, []] }
+
+    assert_nothing_raised do
+      Rack::Lint.new(
+        ActionDispatch::ContentSecurityPolicy::Middleware.new(
+          Rack::Lint.new(app)
+        )
+      ).call(@env)
+    end
+  end
+
+  def test_does_not_override_app_content_security_policy
+    app = proc { [200, { "content-security-policy" => @default_csp }, []] }
+    _, headers, _ = Rack::Lint.new(
+      ActionDispatch::ContentSecurityPolicy::Middleware.new(Rack::Lint.new(app))
+    ).call(@env)
+
+    assert_equal @default_csp, headers["content-security-policy"]
+  end
+
+  def test_does_not_override_app_content_security_policy_for_rack_v2
+    app = proc { [200, { "Content-Security-Policy" => @default_csp }, []] }
+    _, headers, _ = ActionDispatch::ContentSecurityPolicy::Middleware.new(app).call(@env)
+
+    assert_equal @default_csp, headers["content-security-policy"]
+  end
```

Comment:
When using conditional headers, we'll probably have to make this a single test that also uses conditional headers

Reply:
I opted to drop this `v2` test as I don't know how to set the gem version for this test only.
Perhaps that is overthinking it anyway, as these dependencies should be set by the environment.

I did this in my local machine fwiw:

```
% export RACK=2.2.4             
% bundle
# ...

% bundle show rack 
/home/spin/.bundle/rails/gems/rack-2.2.4

% bin/test test/dispatch/content_security_policy_test.rb     
#...
34 runs, 190 assertions, 0 failures, 0 errors, 0 skips           
```